Components: Remove wrapper from global notice rendering

aduth opened the pull request:
This pull request seeks to remove the notice wrapper element added in #3910, as it is not currently in use and causes warnings to appear in the console due to lack of an assigned `key` prop.

[See previous discussion](https://github.com/Automattic/wp-calypso/pull/3910#discussion_r58777946)

__Testing instructions:__

Verify that no regressions occur in the removal of the wrapper element.

/cc @MichaelArestad , @rickybanister

timmyc commented:
The code updates look fine to me and I agree that removing this wrapper and relying upon a modifier class name would work as well.  Not sure if there are some specific steps I can take to visually test this out but it LGTM.

aduth commented:
@timmyc : @rickybanister had recommended copying the markup within the inspector, which I was able to do successfully on the settings screen:

![image](https://cloud.githubusercontent.com/assets/1779930/14789003/7c6b30e0-0ad8-11e6-99af-16d13a305c8f.png)

Looks like there may be some remaining issues with notice styling for non-global notices, since attempting to repeat the same on the sharing connections screen had layout issues similar to the "Before" screenshots in #3910 (not left-aligned, at least).

aduth commented:
Whoops, on a closer look, I wasn't testing the correct branch 😞  In fact, removing the wrapper `<div />` does seem to regress. Will need to spend some more time on this I think...

aduth commented:
@rickybanister : Do you recall why we chose `inline-flex` for notice items?

https://github.com/Automattic/wp-calypso/blob/9257bdd/client/components/global-notices/style.scss#L40

At a glance, changing this to `display: flex;` and removing the wrapper is equally effective at resolving the original issue.

rickybanister commented:
@aduth I asked @MichaelArestad to chime in, I'm not totally sure. We've gone back and forth a lot though, so I want to make sure we don't just ping pong another regression somewhere else.

MichaelArestad commented:
TBH, I don't remember. I was pretty sure the `div` was needed (though couldn't reproduce the why when you asked to remove it) and I think `inline-flex` was needed, but it was done awhile ago. I can do a bit of testing to see if I can figure out why.

mtias commented:
This needs a rebase. @rickybanister @MichaelArestad did you find any regressions during your testing?

aduth commented:
The original issue (key warnings) was resolved in #5506. As such, this PR is more about seeking to remove the wrapper if possible. Otherwise, we could go ahead and close it.

mtias commented:
Removing wrapper if it's not necessary sounds good to me.

rickybanister commented:
testing for regressions might be easier if we had a few more buttons in the devdocs to fire off multiples that stack up, etc. Right now it's tricky to snoop around calyso looking for buttons that trigger the various notices. I'd be happy to work with someone on getting that set up, but would likely need some dev help.

aduth commented:
Rebased this pull request to resolve the merge conflict, and updated the styling to `display: flex;` (from `inline-flex`) to avoid the regression in removing the wrapper component.

Further, I enhanced the `<GlobalNotices />` DevDocs example to add a toggle in switching between creating notices in global (Redux) state, or the legacy `notices` module. This should allow us to more easily test for regressions.

gwwar commented:
I really ❤️  the updated dev docs example. 👍  Tested in Chrome, FF, Safari, IE11 and Edge 🚢

giuseppeg commented:
@aduth I am going to say a bunch of obvious things probably but:
It doesn't matter if they are `inline-flex` or `flex` when `.global-notices`'s position is `fixed`.
When it is not they will behave like block level elements which is great because they will take up all the available space (i.e. they will be as wide as the parent), therefore `flex` is spot on imho.
